game-app/game-core/src/main/java/games/strategy/triplea/ui/ResourceChooser.java

```diff
@@ -0,0 +1,100 @@
+package games.strategy.triplea.ui;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Resource;
+import games.strategy.triplea.Constants;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.swing.DefaultListCellRenderer;
+import javax.swing.JDialog;
+import javax.swing.JList;
+import javax.swing.JOptionPane;
+import javax.swing.ListSelectionModel;
+import org.triplea.swing.SwingComponents;
+
+class ResourceChooser extends JOptionPane {
```

Comment:
I modelled this after PlayerChooser. We could potentially try to make a common base class, but not sure it's worth the extra complexity since the class is relatively short.